pkg/basicserver/basic_server.go

```diff
@@ -38,9 +37,8 @@ type Server interface {
 	GetHTTPClient() *http.Client
 	// AddStartCallback adds a callback in the startServer phase.
 	AddStartCallback(callbacks ...func())
-	// TODO: replace these two methods with `primary` function without etcd server dependency.
-	// GetMember returns the member information.
-	GetMember() *member.Member
-	// AddLeaderCallback adds a callback in the leader campaign phase.
-	AddLeaderCallback(callbacks ...func(context.Context))
+	// IsPrimary returns whether the server is primary, if there is etcd server in the server, it will return whether it is leader.
+	IsPrimary() bool
+	// AddPrimaryCallback adds a callback when the server becomes primary, if there is etcd server in the server, it means the server becomes leader.
```

Comment:
// AddPrimaryCallback adds a callback when the server becomes the leader, if there is embedded etcd, or the primary otherwise.